bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/progress/ProgressAnimationItem.java

```diff
@@ -206,6 +206,26 @@ private void refresh() {
 		if (toolbar == null || toolbar.isDisposed()) {
 			return;
 		}
+		if (bar != null && !bar.isDisposed()) {
+			JobInfo[] jobInfos = ProgressManager.getInstance().getJobInfos(false);
+			int percentSum = 0;
+			int percentCount = 0;
+			for (JobInfo jobInfo : jobInfos) {
+				if (jobInfo != null && !(jobInfo.isBlocked() || jobInfo.getJob().getState() == Job.WAITING)) {
+					int percentDone = jobInfo.getPercentDone();
+					if (percentDone >= 0) {
```

Comment:
if there is currently no progress, should it not still account to the average?

Reply:
percentDone returns -1 for processes that do not report progress. That can not be accounted, The progress is unknown.

Reply:
maybe there should then two counters, one for the ones that report no progress and one for that report progress.

Later then you can set the progress bar to indeterminate state if there are no one reporting actual progress but there are jobs?

Reply:
I would like to but as far as i know it is not possible to change the Style after constructor. There is no ProgressBar.setStyle. :-(

Reply:
using ProgressIndicator to switch between indeterminate. still indeterminate animation looks very unusefull in Windows.